packages/ui/src/views/repo/repo-files/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { ReactNode, useMemo } from 'react'
+
+import { NoData, PathBreadcrumbs, PathParts, SkeletonList } from '@/components'
+import { LatestFileTypes, RepoFile, SandboxLayout } from '@/views'
+import { Summary } from '@/views/repo/components'
+
+interface RepoFilesProps {
+  pathParts: PathParts[]
+  loading: boolean
+  files: RepoFile[]
+  isDir: boolean
+  isShowSummary: boolean
+  latestFile: LatestFileTypes
+  children: ReactNode
+}
+
+export const RepoFiles = ({
+  pathParts,
+  loading,
+  files,
+  isDir,
+  isShowSummary,
+  latestFile,
+  children
+}: RepoFilesProps) => {
+  const content = useMemo(() => {
+    if (!isDir) return children
+
+    if (loading) return <SkeletonList />
+
+    if (isShowSummary && !!files.length) return <Summary latestFile={latestFile} files={files} />
```

Comment:
Not an issue, but you don't need to cast to boolean here
```suggestion
    if (isShowSummary && files.length) return <Summary latestFile={latestFile} files={files} />
```